packages/react-native/ReactCommon/react/renderer/components/view/conversions.h

```diff
@@ -556,6 +556,39 @@ inline void fromRawValue(
   result = transformMatrix;
 }
 
+ inline void fromRawValue(
+     const PropsParserContext &context,
+     const RawValue &value,
+     TransformOrigin &result) {
+  react_native_expect(value.hasType<std::vector<RawValue>>());
+  auto origins = (std::vector<RawValue>)value;
+
+  TransformOrigin transformOrigin;
+
+   for (size_t i = 0; i < 2; i++) {
+     const auto& origin = origins[i];
+     if (origin.hasType<Float>()) {
+       transformOrigin.xy[i] = yogaStyleValueFromFloat((Float)origin);
+     } else if (origin.hasType<std::string>()) {
+       const auto stringValue = (std::string)origin;
+
+       if (stringValue.back() == '%') {
+         auto tryValue = folly::tryTo<float>(
+                                             std::string_view(stringValue).substr(0, stringValue.length() - 1));
+         if (tryValue.hasValue()) {
+           transformOrigin.xy[i] = YGValue{tryValue.value(), YGUnitPercent};
+         }
+       }
+     }
+   }
+
+    if (origins[2].hasType<Float>()) {
+      transformOrigin.z = (Float)origins[2];
+    }
```

Comment:
You do need to check now that the indices you're accessing are not exceeding the vector bounds.
```suggestion
  TransformOrigin transformOrigin;

   for (size_t i = 0; i < std::min(origins.size(), 2); i++) {
     const auto& origin = origins[i];
     if (origin.hasType<Float>()) {
       transformOrigin.xy[i] = yogaStyleValueFromFloat((Float)origin);
     } else if (origin.hasType<std::string>()) {
       const auto stringValue = (std::string)origin;

       if (stringValue.back() == '%') {
         auto tryValue = folly::tryTo<float>(
                                             std::string_view(stringValue).substr(0, stringValue.length() - 1));
         if (tryValue.hasValue()) {
           transformOrigin.xy[i] = YGValue{tryValue.value(), YGUnitPercent};
         }
       }
     }
   }

    if (origin.size() >= 3 && origins[2].hasType<Float>()) {
      transformOrigin.z = (Float)origins[2];
    }
```

Reply:
Updated!